test/integration/nfs-shared-volumes.test.js

```diff
@@ -329,6 +333,127 @@ test('nfs shared volumes', function (tt) {
             });
         });
 
+    tt.test('create a VM that references the volume', function (t) {
+        var payload = {
+            alias: resources.makeResourceName(NFS_SHARED_VOLUMES_VM_NAME_PREFIX,
+                REF_VM_UUID),
+            billing_id: SDC_128_UUID,
+            brand: 'joyent',
+            customer_metadata: {},
+            image_uuid: IMAGE_UUID,
+            networks: [
+                {uuid: ADMIN_OWNED_FABRIC_NETWORK_UUID},
+                {uuid: ADMIN_NETWORK_UUID}
+            ],
+            owner_uuid: UFDS_ADMIN_UUID,
+            uuid: REF_VM_UUID,
+            volumes: [
+                {
+                  name: volumeName,
+                  mode: 'rw',
+                  type: 'tritonnfs',
+                  mountpoint: mountPoint
+                }
+            ]
+        };
+        var user_script = [
+            '#!/bin/bash',
+            '',
+            'cat > /root/.ssh/authorized_keys <<EOF',
+            SSH_PUBLIC_KEY,
+            'EOF',
+            'chmod 0700 /root/.ssh',
+            'chmod 0600 /root/.ssh/authorized_keys',
+            '',
+            'echo "hello" > ' + mountPoint + '/myfile2.txt'
+        ].join('\n');
+
+        payload.customer_metadata['user-script'] = user_script;
+
+        t.comment('creating VM ' + REF_VM_UUID);
+
+        CLIENTS.vmapi.createVmAndWait(payload, {},
+            function onVmCreate(err, job) {
+                t.ifErr(err, 'VM creation should succeed');
```

Comment:
Shouldn't you avoid the later call to `vmapi.getVm` in case we get an error here?

Reply:
Yes - mostly this was copy/paste of other test cases in the file - so just copying the style conventions there - I'll add some better checking.